test/integration/curated_experience_test.go

```diff
@@ -0,0 +1,165 @@
+		matchEndpoints *metadata
+
+		// When dropPod/Endpoint is added, the test will generate a pod/endpoint
+		// with specified metadata and check if is a Dropped Target.
+		dropPod       *metadata
+		dropEndpoints *metadata
+	}{
+		{
+			name: "matching labels",
+			matchPod: &metadata{
+				annotations: map[string]string{"prometheus.io/scrape": "true"},
+				labels:      map[string]string{"app.kubernetes.io/name": "asdTesT1asd"},
+			},
+			matchEndpoints: &metadata{
+				annotations: map[string]string{"prometheus.io/scrape": "true"},
+				labels:      map[string]string{"app.kubernetes.io/name": "asdTesT1asd"},
+			},
+		},
+		{
+			name: "matching labels",
+			matchPod: &metadata{
+				annotations: map[string]string{"prometheus.io/scrape": "true"},
+				labels:      map[string]string{"something-different2": "asdtest2"},
+			},
+			matchEndpoints: &metadata{
+				annotations: map[string]string{"prometheus.io/scrape": "true"},
+				labels:      map[string]string{"something-different2": "asdtest2"},
+			},
+		},
+		{
+			name: "matching annotation without curated",
+			matchPod: &metadata{
+				annotations: map[string]string{"newrelic.io/scrape": "true"},
+			},
+			matchEndpoints: &metadata{
+				annotations: map[string]string{"newrelic.io/scrape": "true"},
+			},
+		},
+		{
+			name: "drop due to curated",
+			dropPod: &metadata{
+				annotations: map[string]string{"prometheus.io/scrape": "true"},
+			},
+			dropEndpoints: &metadata{
+				annotations: map[string]string{"prometheus.io/scrape": "true"},
+			},
+		},
+	}
+
+	for _, testCase := range testsCases {
+		testCase := testCase
+
+		t.Run(testCase.name, func(t *testing.T) {
+			t.Parallel()
+
+			k8sEnv := newK8sEnvironment(t)
+
+			// Resources are only generated when metadata is specified in the test.
+			matchPod := addPodToEnv(t, testCase.matchPod, "match-", k8sEnv)
+			dropPod := addPodToEnv(t, testCase.dropPod, "drop-", k8sEnv)
+			matchEndpointService := addEndpointsToEnv(t, testCase.matchEndpoints, "match-endpoint-", k8sEnv)
+			dropEndpointService := addEndpointsToEnv(t, testCase.dropEndpoints, "drop-endpoint-", k8sEnv)
+
+			nrConfigConfig := fmt.Sprintf(`
+newrelic_remote_write:
+  license_key: nrLicenseKey
+common:
+  scrape_interval: 1s
+kubernetes:
+  jobs:
+    - job_name_prefix: test-k8s
+      target_discovery:
+        endpoints: true
+        pod: true
+        additional_config:
+         kubeconfig_file: %s
+         namespaces:
+          names:
+          - %s
+    - job_name_prefix: newrelic-k8s
+      target_discovery:
+        endpoints: true
+        pod: true
+        additional_config:
+         kubeconfig_file: %s
+         namespaces:
+          names:
+          - %s
+        filter:
+          annotations:
+            newrelic.io/scrape: true
+  curated_experience:
+    enabled: true
+    app_values:
+      - test1
+      - test2
+    source_labels:
+      - something-different1
+      - app.kubernetes.io/name
+      - something-different2
```

Comment:
nit: this names makes a bit difficult to understand the test cases, I would replace them with something like `curatedService` and `notCuratedService`